python/dlr/dlr_model.py

```diff
@@ -120,15 +120,21 @@ def __init__(self, model_path, dev_type='cpu', dev_id=0):
         self.num_inputs = self._get_num_inputs()
         self.num_weights = self._get_num_weights()
         self.input_names = []
+        self.output_names = []
         self.weight_names = []
         self.input_shapes = {}   # Remember shape used in _set_input()
-        for i in range(self.num_inputs):
-            self.input_names.append(self._get_input_name(i))
+        self.input_dtypes = []
+        self.output_dtypes = []
+        
         for i in range(self.num_weights):
             self.weight_names.append(self._get_weight_name(i))
 
         self.num_outputs = self._get_num_outputs()
         self._lazy_init_output_shape()
+        self._fetch_input_names()
+        self._fetch_output_names()
+        self._fetch_input_dtypes()
+        self._fetch_output_dtypes()
```

Comment:
`_fetch_output_dtypes` and `_fetch_input_dtypes` will work for TVM, Treelite and TFLite backends. Other backends do not support it yet.
So, we should have try catch here as well

Reply:
👍